Formula/e/egctl.rb

```diff
@@ -0,0 +1,30 @@
+class Egctl < Formula
+  desc "Command-line utility for operating Envoy Gateway"
+  homepage "https://gateway.envoyproxy.io/"
+  url "https://github.com/envoyproxy/gateway/archive/refs/tags/v1.0.2.tar.gz"
+  sha256 "05406182dc165513925cf60722582613d4de9ea789d60e014e6da456bb229f65"
+  license "Apache-2.0"
+  head "https://github.com/envoyproxy/gateway.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/envoyproxy/gateway/internal/cmd/version.envoyGatewayVersion=#{version}
+      -X github.com/envoyproxy/gateway/internal/cmd/version.gitCommitID=#{tap.user}
+    ]
+    system "go", "build", *std_go_args(ldflags:), "./cmd/egctl"
+
+    generate_completions_from_executable(bin/"egctl", "completion")
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/egctl version --remote=false").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks, updated!